tark/tark/utils/exon_utils.py

```diff
@@ -85,137 +85,38 @@ def compute_cds_sequence(cls, transcript):
-                    if overlap_score > 0:
-                        end_exon = exon
-                        break
-                #
-                #                 #  collect all the previous exons to calculate the three prime utr length
-                #                 #  Work here tomorrow
-                three_prime_exon_iterator = iter(exons_copy)
-                current_exon = next(three_prime_exon_iterator)
-                end_exon_order = end_exon['exon_order']
-
-                # print('End exon order ' + str(end_exon_order))
-                while (current_exon['exon_order'] > end_exon_order):
-                    # print(' 3 prime UTR exon %d' %current_exon['exon_order'])
-                    three_prime_utr_seq = current_exon['sequence'] + three_prime_utr_seq
-                    current_exon = next(three_prime_exon_iterator)
-
-                # Now add the overlapping bit for the 3 prime
-                if transcript['loc_strand'] == -1:
-                    end_exon_utr_len = translation_start - end_exon['loc_start']
-                    if end_exon_utr_len > 0:
-                        three_prime_utr_seq = end_exon['sequence'][-end_exon_utr_len:] + three_prime_utr_seq
-                else:
-                    end_exon_utr_len = end_exon['loc_end'] - translation_end
-                    if end_exon_utr_len > 0:
-                        three_prime_utr_seq = end_exon['sequence'][-end_exon_utr_len:] + three_prime_utr_seq
-
-            if start_exon is not None:
-                cds_start_len = translation_start - start_exon['loc_start']
-                cds_seq = start_exon['sequence'][cds_start_len:]
-
-            if exons is not None:
-                last_exon = exons[-1]
-
-                if transcript['loc_strand'] == -1:
-                    if len(three_prime_utr_seq) > 0:
-                        cds_info['three_prime_utr_end'] = last_exon['loc_start']
-                    else:
-                        cds_info['three_prime_utr_start'] = 0
-                        cds_info['three_prime_utr_end'] = 0
-                else:
-                    if len(three_prime_utr_seq) > 0:
-                        cds_info['three_prime_utr_end'] = last_exon['loc_end']
-                    else:
-                        cds_info['three_prime_utr_start'] = 0
-                        cds_info['three_prime_utr_end'] = 0
-
-            cds_info['five_prime_utr_seq'] = five_prime_utr_seq
-            cds_info['five_prime_utr_length'] = len(five_prime_utr_seq)
-            cds_info['three_prime_utr_seq'] = three_prime_utr_seq
-            cds_info['three_prime_utr_length'] = len(three_prime_utr_seq)
+            cds_info["translation_start"] = translation["loc_start"]
+            cds_info["translation_end"] = translation["loc_end"]
+            cds_info["three_prime_utr_start"] = ExonUtils.get_or_default(transcript, "three_prime_utr_start", 0)
```

Comment:
Do we need `ExonUtils.get_or_default()` method? Doesn't transcript.get('three_prime_utr_start', 0) do the same functionality?

Reply:
I don't want to use `get` here because `get` returns the default value only if the requested key does not exist in the dictionary.  I want to set a default value of `0` even when the key does exist in the dictionary but has a value of `None`, which often happens in this case because `NULL` entries in the `three_prime_utr_start` column are mapped to `None` by Django.  I'll add a comment to clarify this.

Reply:
Makes sense. Thank you.